### ISSUE TITLE
Demo Requires Reinstall/Reboot to Relaunch

### ISSUE THREAD

**Issue by jam1garner** [leetvr/hotham#134]
## Description

**Behavior observed:** After first running the demo either a reinstall of the demo or a reboot of the quest is required to launch again. Additional launches result in an unrendered window (eg an empty "2d android app" window pops up, but the "contents" are just transparent/nonexistant). No Vulkan errors in the logs and they mostly seem to be exactly how they look when the demo works properly.

I have a suspicion this bit of the logs is indirectly relevant:

```
01-27 10:54:23.187 13791 14139 I RustStdoutStderr: thread '<unnamed>' panicked at 'Failed to bind: Os { code: 98, kind: AddrInUse, message: "Address already in use" }', hotham-debug-server/src/lib.rs:116:43
01-27 10:54:23.187 13791 14139 I RustStdoutStderr: stack backtrace:
01-27 10:54:23.188 13791 14139 I RustStdoutStderr: note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

If I had to guess, your android event handling isn't properly exiting the process, so when it tries to launch a second instance it tries to begin debugging but the address is already bound to by the activity. I looked through your code and am honestly having a bit of trouble finding any issue in how you're handling this.

My one guess is that this unconditional break here seems incorrect:

https://github.com/leetvr/hotham/blob/ea8c0b254f07c464fd9885b5cd6ddaf5d47fbc7e/hotham/src/app.rs#L110-L124

I think what you actually want here is:

```rust
            while let Some(event) = self.poll_android_events() {
                println!("[HOTHAM_ANDROID] Received event {:?}", event);
                match event {
                    ndk_glue::Event::Resume => self.resumed = true,
                    ndk_glue::Event::Destroy | ndk_glue::Event::WindowDestroyed => {
                        self.should_quit.store(true, Ordering::Relaxed);
                        return true;
                    }
                    ndk_glue::Event::Pause => self.resumed = false,
                    _ => {}
                }
            }
```

eg rather than break after the first iteration, stop looping once there are no more events to handle. I also can't say I have the highest confidence in the empty wildcard branch either.

## Details
- Quest 2, Latest FW (v37)

**Comment by jam1garner** [comment]
I'll try and test/PR my fix soon

**Comment by kanerogers** [comment]
This should be fixed as of #138 🤞🏻
